logs/trainrun_forest/config.txt
lr >>> 0.0001

dropout >>> 0.0

emsize >>> 512

batch_size >>> 4

nlayers >>> 12

num_features >>> 100

nhead >>> 4

nhid_factor >>> 2

bptt >>> 1152

eval_positions >>> [1094]

seq_len_used >>> 50

sampling >>> normal

epochs >>> 50

num_steps >>> 640

verbose >>> False

mix_activations >>> True

pre_sample_causes >>> True

multiclass_type >>> rank

nan_prob_unknown_reason_reason_prior >>> 0.5

categorical_feature_p >>> 0

nan_prob_no_reason >>> 0.0

nan_prob_unknown_reason >>> 0.0

nan_prob_a_reason >>> 0.0

max_num_classes >>> 2

num_classes >>> 2

noise_type >>> Gaussian

balanced >>> True

normalize_to_ranking >>> False

set_value_to_nan >>> 0.0

normalize_by_used_features >>> True

num_features_used >>> 100

num_categorical_features_sampler_a >>> -1.0

differentiable_hyperparameters >>> {'prior_bag_exp_weights_1': {'distribution': 'uniform', 'min': 2.0, 'max': 10.0}, 'num_layers': {'distribution': 'meta_gamma', 'max_alpha': 2, 'max_scale': 3, 'round': True, 'lower_bound': 2}, 'prior_mlp_hidden_dim': {'distribution': 'meta_gamma', 'max_alpha': 3, 'max_scale': 100, 'round': True, 'lower_bound': 4}, 'prior_mlp_dropout_prob': {'distribution': 'meta_beta', 'scale': 0.6, 'min': 0.1, 'max': 5.0}, 'noise_std': {'distribution': 'meta_trunc_norm_log_scaled', 'max_mean': 0.3, 'min_mean': 0.0001, 'round': False, 'lower_bound': 0.0}, 'init_std': {'distribution': 'meta_trunc_norm_log_scaled', 'max_mean': 10.0, 'min_mean': 0.01, 'round': False, 'lower_bound': 0.0}, 'num_causes': {'distribution': 'meta_gamma', 'max_alpha': 3, 'max_scale': 7, 'round': True, 'lower_bound': 2}, 'is_causal': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'pre_sample_weights': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'y_is_effect': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'prior_mlp_activations': {'distribution': 'meta_choice_mixed', 'choice_values': [<class 'torch.nn.modules.activation.Tanh'>, <class 'torch.nn.modules.linear.Identity'>, <class 'torch.nn.modules.activation.ReLU'>]}, 'block_wise_dropout': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'sort_features': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'in_clique': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'outputscale': {'distribution': 'meta_trunc_norm_log_scaled', 'max_mean': 10.0, 'min_mean': 1e-05, 'round': False, 'lower_bound': 0}, 'lengthscale': {'distribution': 'meta_trunc_norm_log_scaled', 'max_mean': 10.0, 'min_mean': 1e-05, 'round': False, 'lower_bound': 0}, 'noise': {'distribution': 'meta_choice', 'choice_values': [1e-05, 0.0001, 0.01]}}

prior_type >>> forest

recompute_attn >>> True

differentiable >>> True

flexible >>> True

bptt_extra_samples >>> None

output_multiclass_ordered_p >>> 0.0

multiclass_loss_type >>> nono

new_mlp_per_example >>> True

prior_mlp_scale_weights_sqrt >>> True

batch_size_per_gp_sample >>> None

differentiable_hps_as_style >>> False

max_eval_pos >>> 1000

random_feature_rotation >>> True

rotate_normalized_labels >>> True

canonical_y_encoder >>> False

total_available_time_in_s >>> None

train_mixed_precision >>> False

efficient_eval_masking >>> True

is_causal >>> True

num_causes >>> 5

prior_mlp_hidden_dim >>> 50

num_layers >>> 4

noise_std >>> 0.05

init_std >>> 0.05

y_is_effect >>> True

pre_sample_weights >>> True

prior_mlp_dropout_prob >>> 0

prior_mlp_activations >>> <class 'torch.nn.modules.activation.ReLU'>

block_wise_dropout >>> True

sort_features >>> False

in_clique >>> False

max_features >>> 100

min_features >>> 100

n_samples >>> 1000

base_size >>> 1000

n_estimators >>> 1

min_depth >>> 1

max_depth >>> 15

categorical_x >>> False

data_sample_func >>> mnd

comp >>> False

no_encoder >>> False

normalize_with_sqrt >>> False

normalize_ignore_label_too >>> False

normalize >>> True

microbiome_test >>> True

weight_classes >>> False

run_name >>> forest

aggregate_k_gradients >>> 16



logs/trainrun_mlp_multi/config.txt
lr >>> 1e-06

dropout >>> 0.0

emsize >>> 512

batch_size >>> 4

nlayers >>> 12

num_features >>> 100

nhead >>> 4

nhid_factor >>> 2

bptt >>> 1152

eval_positions >>> [1094]

seq_len_used >>> 50

sampling >>> normal

epochs >>> 50

num_steps >>> 64

verbose >>> False

mix_activations >>> True

pre_sample_causes >>> True

multiclass_type >>> rank

nan_prob_unknown_reason_reason_prior >>> 0.5

categorical_feature_p >>> 0

nan_prob_no_reason >>> 0.0

nan_prob_unknown_reason >>> 0.0

nan_prob_a_reason >>> 0.0

max_num_classes >>> 2

num_classes >>> 10

noise_type >>> Gaussian

balanced >>> False

normalize_to_ranking >>> False

set_value_to_nan >>> 0.0

normalize_by_used_features >>> True

num_features_used >>> 100

num_categorical_features_sampler_a >>> -1.0

differentiable_hyperparameters >>> {'prior_bag_exp_weights_1': {'distribution': 'uniform', 'min': 2.0, 'max': 10.0}, 'num_layers': {'distribution': 'meta_gamma', 'max_alpha': 2, 'max_scale': 3, 'round': True, 'lower_bound': 2}, 'prior_mlp_hidden_dim': {'distribution': 'meta_gamma', 'max_alpha': 3, 'max_scale': 100, 'round': True, 'lower_bound': 4}, 'prior_mlp_dropout_prob': {'distribution': 'meta_beta', 'scale': 0.6, 'min': 0.1, 'max': 5.0}, 'noise_std': {'distribution': 'meta_trunc_norm_log_scaled', 'max_mean': 0.3, 'min_mean': 0.0001, 'round': False, 'lower_bound': 0.0}, 'init_std': {'distribution': 'meta_trunc_norm_log_scaled', 'max_mean': 10.0, 'min_mean': 0.01, 'round': False, 'lower_bound': 0.0}, 'num_causes': {'distribution': 'meta_gamma', 'max_alpha': 3, 'max_scale': 7, 'round': True, 'lower_bound': 2}, 'is_causal': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'pre_sample_weights': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'y_is_effect': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'prior_mlp_activations': {'distribution': 'meta_choice_mixed', 'choice_values': [<class 'torch.nn.modules.activation.Tanh'>, <class 'torch.nn.modules.linear.Identity'>, <class 'torch.nn.modules.activation.ReLU'>]}, 'block_wise_dropout': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'sort_features': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'in_clique': {'distribution': 'meta_choice', 'choice_values': [True, False]}, 'outputscale': {'distribution': 'meta_trunc_norm_log_scaled', 'max_mean': 10.0, 'min_mean': 1e-05, 'round': False, 'lower_bound': 0}, 'lengthscale': {'distribution': 'meta_trunc_norm_log_scaled', 'max_mean': 10.0, 'min_mean': 1e-05, 'round': False, 'lower_bound': 0}, 'noise': {'distribution': 'meta_choice', 'choice_values': [1e-05, 0.0001, 0.01]}}

prior_type >>> mlp

recompute_attn >>> True

differentiable >>> True

flexible >>> True

bptt_extra_samples >>> None

output_multiclass_ordered_p >>> 0.0

multiclass_loss_type >>> nono

new_mlp_per_example >>> True

prior_mlp_scale_weights_sqrt >>> True

batch_size_per_gp_sample >>> None

differentiable_hps_as_style >>> False

max_eval_pos >>> 1000

random_feature_rotation >>> True

rotate_normalized_labels >>> True

canonical_y_encoder >>> False

total_available_time_in_s >>> None

train_mixed_precision >>> False

efficient_eval_masking >>> True

is_causal >>> True

num_causes >>> 5

prior_mlp_hidden_dim >>> 50

num_layers >>> 4

noise_std >>> 0.05

init_std >>> 0.05

y_is_effect >>> True

pre_sample_weights >>> True

prior_mlp_dropout_prob >>> 0

prior_mlp_activations >>> <class 'torch.nn.modules.activation.ReLU'>

block_wise_dropout >>> True

sort_features >>> False

in_clique >>> False

max_features >>> 100

min_features >>> 100

n_samples >>> 1000

base_size >>> 1000

n_estimators >>> 1

min_depth >>> 1

max_depth >>> 25

categorical_x >>> False

data_sample_func >>> mnd

comp >>> False

no_encoder >>> False

normalize_with_sqrt >>> False

normalize_ignore_label_too >>> False

normalize >>> False

microbiome_test >>> True

weight_classes >>> False

run_name >>> mlp_multi

aggregate_k_gradients >>> 16



tabularbench/models/TabSurvey/models/dnf_lib/config.py
import os
import copy

from sklearn.metrics import mean_squared_error

from tabularbench.models.dnf_lib.Utils.file_utils import create_dir


my_config = {
    'n_formulas': 64,
    'orthogonal_lambda': 0.,
    'elastic_net_beta': 0.1,
    'random_seed': 1306,
    
    'input_dim': 54,
    'output_dim': 7,
    'translate_label_to_one_hot': True,

    # 'XGB_objective': "multi:softprob", #"binary:logistic",
    
    'experiment_number': 1,

    'GPU': '0'
}

# Values for hyperparameter search
DNFNet_grid_params = {
    'n_formulas': [3072, 2048, 1024, 512, 256, 128, 64],
    'orthogonal_lambda': [0.],
    'elastic_net_beta': [1.6, 1.3, 1., 0.7, 0.4, 0.1],
}


shared_config = {
    'n_conjunctions_arr': [6, 9, 12, 15],
    'conjunctions_depth_arr': [2, 4, 6],
    'keep_feature_prob_arr': [0.1, 0.3, 0.5, 0.7, 0.9],

    'initial_lr': 5e-2,
    'lr_decay_factor': 0.5,
    'lr_patience': 10,
    'min_lr': 1e-6,

    'early_stopping_patience': 20,
    'epochs': 1000,
    'batch_size': 128, #2048,

    'apply_standardization': True,

    'save_weights': True,
    'starting_epoch_to_save': 0,
    'models_module_name': 'models.dnf_lib.DNFNet.DNFNetModels',
    'models_dir': 'models/dnf_lib/DNFNet/DNFNetModels',
    
    'model_number': 1  # That is the whole DNF with localization and feature selection
}

score_config = {
    'score_metric': mean_squared_error,
    'score_increases': False,
    # 'XGB_eval_metric': 'acc',
}


def merge_configs(config_dst, config_src):
    config_dst = copy.deepcopy(config_dst)
    config_src = copy.deepcopy(config_src)
    for key, value in config_src.items():
        config_dst[key] = value
    return config_dst


def get_config(competition_name, model):
    
    shared_config['experiments_dir'] = 'output/DNFNet/experiments/{}'.format(competition_name)
    output_dir = os.path.join(shared_config['experiments_dir'], 'grid_search')
    create_dir(shared_config['experiments_dir'])
    shared_config['competition_name'] = competition_name
    shared_config['model_name'] = model
    
    config = merge_configs(shared_config, my_config)
    create_dir(output_dir)
        
    return config, score_config
